### ISSUE TITLE
powerMonitor.getSystemIdleTime() reports user idle on linux when using only keyboard (on Wayland)

### ISSUE THREAD

**Issue by ssalenik** [electron/electron#27912]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
10.0.2
https://github.com/johannesjo/super-productivity/blob/master/package.json#L145

* **Operating System:**
Ubuntu 20.10 x64 Wayland

* **Last Known Working Electron version:**
N/A

### Expected Behavior
When user is using only a keyboard (eg: editing files in Vim in running in a terminal emulator) `powerMonitor.getSystemIdleTime()` API does not detect this as idle time.

### Actual Behavior
`powerMonitor.getSystemIdleTime()` detects the user is idle when user is using keyboard only in Vim

### To Reproduce
Found this while using `Super-Productivity` app on latest release version `6.4.0` from Snap store. Bug filed here:
https://github.com/johannesjo/super-productivity/issues/922

This app is using the `powerMonitor` API to detect when a user is idle to notify them after X number of minutes:
https://github.com/johannesjo/super-productivity/blob/03ff2fb0086225b9b56832e308b2751e2afc0f52/electron/main.ts#L146

Using the app I get this idle detection whenever I'm using the keyboard only in Vim for X number of minutes.

**Comment by ssalenik** [comment]
After some more investigation, I believe this might be related to Wayland. Seems that most idle detection methods on Linux used XScreenSaver or something similar relying on X which probably doesn't work on Wayland.

**Comment by johannesjo** [comment]
Still relevant

**Comment by hovancik** [comment]
still issue electron 22

**Comment by juliankoehn** [comment]
I am experiencing the same issue with Ubuntu Core 22 and Wayland. I am using Electron 22. Unfortunately, I am always receiving 'active' as the idle state. Has anyone found a workaround for this problem on Ubuntu Core 22 and Electron 22?

**Comment by maymage** [comment]
.

**Comment by hovancik** [comment]
still issue

**Comment by hovancik** [comment]
still issue with 25, please remove stale

**Comment by hovancik** [comment]
Bump

**Comment by meenakshi-koushik** [comment]
I can easily reproduce this problem with electron v28 (and probably also with earlier versions) using electron-quick-start. By periodically logging `powerMonitor.getSystemIdleTime()` in main.js. When I run the app in X, the idle time reported increases monotonously as soon as I stop using keyboard/ mouse, while in Wayland (ubuntu 22.04,  gnome) the idle time is always reported as 0.

**Comment by wolftune** [comment]
incidentally, I have heard there are multiple idle-time protocols in Wayland. KDE (kwin) has a different one from some other frameworks or something. I hope they will all be supported.

**Comment by mcmxcdev** [comment]
Bump

**Comment by mcmxcdev** [comment]
Bump

**Comment by Rdkang** [comment]
Bump

**Comment by johannesjo** [comment]
bump

**Comment by llyyr** [comment]
> incidentally, I have heard there are multiple idle-time protocols in Wayland. KDE (kwin) has a different one from some other frameworks or something. I hope they will all be supported.

Chromium supports the KDE and GNOME specific protocols, but for some reason they don't work in Electron.

Chromium for example will subscribe to kwin_idle if available:

```
[3599877.748]  -> org_kde_kwin_idle#11.get_idle_timeout(new id org_kde_kwin_idle_timeout#38, wl_seat#28, 5000)
```

But Electron never does so. There's also a Chromium ticket for supporting the platform agnostic protocol https://issues.chromium.org/issues/380125108 but it's unlikely if this will fix this issue for Electron.

**Comment by codebytere** [comment]
This is the underlying implementation: https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform/wayland/host/wayland_screen.cc;l=463-485;drc=4c6d5b3c3ed16fa75258c6b4763188f840cb433d;bpv=1;bpt=1. We delegate directly to it via `ui::CalculateIdleTime()` so there's unfortunately nothing to be done about it on our end at the moment. My best recommendation is to agitate a little about it upstream in https://issues.chromium.org/issues.

**Comment by llyyr** [comment]
> My best recommendation is to agitate a little about it upstream in https://issues.chromium.org/issues.

Why would it be a Chromium bug if it works on Chromium but not on Electron?

**Comment by MomoRazor** [comment]
@codebytere Sorry to butt in, but would it be possible to point us towards the specific issue on Chromium we can bump on, or has one not yet been created? Thank you!

**Comment by hovancik** [comment]
Hi @MomoRazor , as you don't have issues enabled in your package, trying here: 

I've managed to install the package after installing dev packages for libinput and libudev, but can't find usage instructions. If I try to follow original instractions before your fork `require('desktop-idle').getIdleTime()`, I am getting error `Error: Cannot find module 'desktop-idle'` .Would be nice to include instructions in readme :) Also, I think link/name of the npm package would be also useful for people. 

Thanks!

**Comment by MomoRazor** [comment]
Hi @hovancik, sorry for not having issues enabled, my bad! They should be enabled so we can continue the conversation there :).

However in way of reply, the following is a link and name to the npm package:
- Link: https://www.npmjs.com/package/node-desktop-idle-v2
- Name: node-desktop-idle-v2

As for instructions, you are a 100% correct, I neglected to add proper instructions as  I thought the older instructions would be enough, but that stopped being true fairly quick in the upgrade honestly, sorry. I'll be adding proper instructions asap. 

For some quick support, since the node-desktop-idle-v2 is now an ESModule, it should be imported as following:

`import { desktopIdle } from 'node-desktop-idle-v2'`

As to get the idle time, it should still be:

`desktopIdle.getIdleTime()`

If this does not work as expected, feel free to open an issue on the repo and I'll help you asap :). Thank you!

**Comment by llyyr** [comment]
@MomoRazor your package can't work for linux desktops without giving your application root privileges. /dev/input is restricted to root users only.

I'd suggest restoring the previous X11 solution for X11 environments, and supporting https://wayland.app/protocols/ext-idle-notify-v1 for Wayland

**Comment by MomoRazor** [comment]
In order to minimise clutter in this thread, I've created an issue on the package's page for further discussion:

https://github.com/MomoRazor/node-desktop-idle-v2/issues/1

TLDR; I've provided ways for users to simply be given /dev/inputs permissions without being promoted to or requiring root access. However, due to my relative ignorance in linux systems, I'm open to feedback as to the security implications of this, and hence the merits of re-implementing a different approach. Thank you! :)
